### PR TITLE
Use pygithub instead of raw http requests, convert to library

### PR DESCRIPTION
**Pull request by noxasaxon**
**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: socless_repo_parser/builders.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    integration_family.meta.integration_family = repo_name_info.name
+
+    integration_family.meta.repo_url = repo_name_info.url
+
+    # get repo object
+    gh_repo = github_instance.get_repo(f"{repo_name_info.org}/{repo_name_info.name}")
+
+    # get serverless.yml
+    file_contents = gh_repo.get_contents(SERVERLESS_YML)
+    assert not isinstance(file_contents, list)
```
Comment (ubaniabalogun) [inline socless_repo_parser/builders.py]: #question: Is the assert here being used as a debugging aid during development? Or is there a need to check at runtime that file_contents is not an instance of `list`? (a check that shouldn't be done with `assert`)

File: socless_repo_parser/builders.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                    repo_meta, self.get_or_init_github_enterprise()
+                )
+            else:
+                integration_family = build_integration_family(
+                    repo_meta, self.get_or_init_github()
+                )
+
+            all_integrations.integrations.append(integration_family)
+
+        return all_integrations
```
Comment (ubaniabalogun) [inline socless_repo_parser/builders.py]: #generalcomment: This is a general comment, i.e. no changes required at this point, but something to mull on

I'd like us to revisit the implementation of this class in the next iteration of this codebase. There seems to be a lot of similarities between the `*_github` functions and the `*_github_enterprise` functions, albeit some small differences. My gut feel is that we might be able to "lift out" the similarities and isolate the differences in a way that leads to less code and increased legibility. I'll cut an issue for this in the github so we remember and we can brainstorm some alternate implementations.

File: tests/test_parse_yml.py [hunk 3]
```diff
@@ -5,4 +5,4 @@ def test_parse_yml():
     with open("tests/mock_files/mock_serverless.yml") as f:
         mock_serverless_yml_as_string = f.read()
 
-    result = parse_yml(mock_serverless_yml_as_string)
+    _ = parse_yml(mock_serverless_yml_as_string)
```
Comment (ubaniabalogun) [inline tests/test_parse_yml.py]: Why not lose the assignment all together since its unused?

Comment (ubaniabalogun) [reply tests/test_parse_yml.py]: Also, what does this test assert?

File: socless_repo_parser/builders.py (hunk 1 above)
Comment (noxasaxon) [reply socless_repo_parser/builders.py]: the `get_contents` method can return either a list (if the path points to a directory) or a single item (if the path points to a file). For fetching the serverless_yml path, that should not be a directory so if it pulls back more than one item we need the program to fail immediately

File: tests/test_parse_yml.py (hunk 3 above)
Comment (noxasaxon) [reply tests/test_parse_yml.py]: updated test

File: socless_repo_parser/builders.py (hunk 1 above)
Comment (noxasaxon) [reply socless_repo_parser/builders.py]: updated the asserts to raise specific custom library errors for better error messaging

**Review by ubaniabalogun — APPROVED** [review]

**Pull request merged**